generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs

```diff
@@ -1405,6 +1405,59 @@ class <%= entityClass %>ResourceIT <% if (databaseType === 'cassandra') { %>exte
         <%_ } _%>
     }
 
+    @Test<%= transactionalAnnotation %>
+    void partialUpdate<%= entityClass %>() throws Exception {
+        // Initialize the database
+        <%_ if (databaseType === 'cassandra') { _%>
+            <%= asEntity(entityInstance) %>.setId(UUID.randomUUID());
+        <%_ } _%>
+        <%= entityInstance %>Repository.<%= saveMethod %>(<%= asEntity(entityInstance) %>)<%= callBlock %>;
+
+        int databaseSizeBeforeUpdate = <%= entityInstance %>Repository.findAll()<%= callListBlock %>.size();
+
+        // Update the <%= entityInstance %> using partial update
+        <%= asEntity(entityClass) %> partialUpdated<%= asEntity(entityClass) %> = <%= entityInstance %>Repository.findById(<%= asEntity(entityInstance) %>.getId()).get() %>;
+        <%_ if (databaseType === 'sql') { _%>
+        // Disconnect from session so that the updates on updated<%= asEntity(entityClass) %> are not directly saved in db
+        em.detach(partialUpdated<%= asEntity(entityClass) %>);
+        <%_ } _%>
```

Comment:
Why don't create the instance directly instead of loading and detaching?
```
<%= asEntity(entityClass) %> partialUpdated<%= asEntity(entityClass) %> = new <%= asEntity(entityClass) %>();
partialUpdated<%= asEntity(entityClass) %>.setId(<%= asEntity(entityInstance) %>.getId());
```

Reply:
And I am missing a test with a not existing id.

Reply:
@mshima the first reason why we did this is because it's done in the same way when updating the entity (Line 1319 in the same file).
If we dive into this, I think that the detach has been added to prevent getting "wrong working tests":
If you get an entity, then edit its fields, you editing the reference fields and you could get a test that is working just because the reference has been changed in the test, not because the Patch is working correctly.

I already encountered an issue like this:
 - I was getting an entity, lets say `User user1`
 - I was editing this entity: `user->setName("test")`
 - Then calling the repository to get all the entities with name `test`: `userRepository.findByName("test")`
 - And the entity I just got and edited **without saving** was returned from `findByName`

I can't be sure that it's the reason why it's been added to the update method, but to my mind, it would be safer to keep it. What do you think?

Reply:
The update method is correct, we should load the entity so we don't miss and required field/relationship.
In this case I don't see the reason to load the entity, just create an empty instance.
PATCH should persist only changed fields.
When we load a persisted entity, edit it and save it, the workflow is identical to existing update method.
The old values will be passed to the patch method, so there is no guarantee if patch is loading and persisting not modified fields correctly.

IMO the correct approach would be to have 2 tests:
- Partial patch update
  - create an empty instance
  - set an updated value to half of the fields
  - execute patch.
  - load the updated entity
  - compare all field values, updated against updated value and non updated fields against old value.
- A complete patch update.
  - same as partial but with every field for completeness.

IMO the correct approach would be to use sql update with criteria api, but this would complicate non sql, and others use cases.

Reply:
I understand, I'll develop the two different tests. We mentionned this when we were at JHipster code, but we forgot to do this :)

Reply:
I'll also add a test with a missing id, to have a full coverage

Reply:
I think the easier way is to extract your partialUpdate test to another template and use something like:
```
<% const fieldsToIncludeInPartialTest = fields.map(field => {
    const includeField = faker.random.boolean();
    const updateWithValue = includeField ? `UPDATED_${}` : 'null';
    const testWithConstant = includeField ? `UPDATED_${}` : `DEFAULT_${}`;
    return { includeField, updateWithValue, testWithConstant, field};
}); %>
<%- include('partial_it_test`, {fields: fieldsToIncludeInPartialTest}) -%>
```

Reply:
Thank you very much for the advice, I just pushed the 3 tests.
I created the template in `tests/java/package/common`, is it ok for you? There was no folder to store templates for the tests.
Here is the commit link: https://github.com/jhipster/generator-jhipster/pull/12507/commits/c1927270545d7de526e5b75df0ba5f93b11c9b88

Reply:
At first glance I would think `templates/src/tests/java/package/common` should go to `src/tests/java/package/common`.
I prefer:
- put in the same folder as the main file since it’s used only there
- put inside `templates/partials` and name it with a complete description like `it_patch_update.partial.java.ejs`.

Reply:
I choosed the `templates/partials` solution: https://github.com/jhipster/generator-jhipster/pull/12507/commits/7f859d2cfa3358cea044683640e31f075ff917bc
Does it seems ok, or is there a bettery way than doing `../../......`?

Reply:
> Does it seems ok, or is there a bettery way than doing `../../......`?

You can try adding.
```
    options = {
        root: generator.templatePath(),
        context: generator,
        ...options,
    };
```
to https://github.com/jhipster/generator-jhipster/blob/ad3e0fbc48a21369444595cf0c6d54d8151a28bb/generators/utils.js#L240

This should allow to include the file using absolute path (relative to templates path) `/partials/...ejs`
https://ejs.co/#docs

Reply:
Awesome, I changed this and it works, thank you!

Reply:
Great